src/plugins/dashboard/public/application/components/dashboard_no_match.tsx

```diff
@@ -3,8 +3,21 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-import React from 'react';
+import { useEffect } from 'react';
+import { useOpenSearchDashboards } from '../../../../opensearch_dashboards_react/public';
+import { DashboardServices } from '../../types';
 
 export const DashboardNoMatch = () => {
-  return <div>Dashboard No Match</div>;
+  const { services } = useOpenSearchDashboards<DashboardServices>();
+  useEffect(() => {
+    const path = window.location.hash.substr(1);
```

Comment:
`substr` is deprecated.

```suggestion
    const path = window.location.hash.substring(1);
```

Reply:
fixed